arangod/Aql/SortBlock.cpp

```diff
@@ -395,12 +411,15 @@ class ConstrainedHeapSorter : public arangodb::aql::SortBlock::Sorter {
         return {res, TRI_ERROR_NO_ERROR};
       }
 
+      if (!_buffer.empty()) {
+        ensureHeapBuffer(_buffer.front());  // make sure we have a dst
+      }
       // handle batch
       while (!_buffer.empty()) {
         std::unique_ptr<AqlItemBlock> src{_buffer.front()};
+        _cmpInput.setBuffers(_heapBuffer.get(), src.get());
         auto cleanup =
```

Comment:
do we really need a cleanup function here?
Couldn't we simply call `_buffer.pop_front()` after making the unique_ptr take over the ownership?
Or is `_buffer` used inside `pushRow()`?